### ISSUE TITLE
end-tag rule ignores HTML LS tag-omission rules, over-detects on li/p/td/option etc.

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3965]
## Summary

The `end-tag` rule (`packages/@markuplint/rules/src/end-tag/index.ts`) reports "Missing the end tag" for *every* element whose `closeTag` is `null`, without checking HTML LS's [tag omission rules](https://html.spec.whatwg.org/multipage/syntax.html#optional-tags). This causes over-detection on elements whose end tag is legally omittable (`<li>`, `<p>`, `<td>`, `<tr>`, `<option>`, `<thead>`/`<tbody>`/`<tfoot>`, `<dt>`/`<dd>`, `<colgroup>`, `<caption>`, `<rt>`/`<rp>`, `<optgroup>`, `<head>`/`<body>`/`<html>`), and — more importantly for this Issue — makes it unsafe to add the rule to the nu-validator coverage bench with `severity: 'error'`, because on any fixture using one of the above (extremely common in real-world and test HTML) it would produce a false positive that the spec explicitly permits.

The rule's own README already documents this as a known limitation:

> Currently, this rule doesn't evaluate whether omit possible the end tag.

## Repro (over-detection on legally-omittable end tags)

```html
<ul><li>foo<li>bar</ul>
```
→ `end-tag` reports "Missing the end tag" on both `<li>` occurrences, even though HTML LS explicitly permits omitting `</li>` when immediately followed by another `<li>` or the parent's end tag.

Verified empirically (`mlRuleTest`) against `<li>`, `<p>`, `<td>`, `<option>` — all four report a spurious violation.

## Repro (the actual coverage gap this blocks)

```html
<picture><img src=x alt></picture>
```
(end tag omitted entirely, and `<picture>` is not in HTML LS's omission list — [§4.8.1](https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element): "Tag omission in text/html: Neither tag is omissible.")

nu-validator: error — `Unclosed element "picture".` / `End of file seen and there were open elements.`

markuplint's `end-tag` rule *does* fire correctly here (picture's end tag is never legally omittable, so the naive "closeTag is null → violation" check happens to be right) — but it fires at `defaultSeverity: 'warning'`, and the rule's category is `'style'`, not a spec-conformance category, precisely because of the false-positive risk documented above. `compare.ts` only counts `severity === 'error'` violations, so this fixture stays `nu-only` even with the rule enabled.

Fixture: `tests/external/validator/tests/html/elements/picture/html-syntax-picture-no-end-tag-novalid.html` (`nu-only`).

## Why a blanket severity override is unsafe

A naive fix — add `'end-tag': { severity: 'error' }` to `tests/external/bench/config.ts` — was considered and rejected during triage: it would flip every fixture in the ~5400-file corpus that legally omits `<li>`/`<p>`/`<td>`/`<option>`/etc. end tags from `match-clean` to `ml-only`, which is a genuine markuplint false positive, not a "ml strict, nu lax" case (see `.claude/skills/bench-triage/SKILL.md`'s "ml-only readings" classification). That is a real regression, not a triage artifact.

## Proposed direction

Make `end-tag` omission-table-aware:

1. Build (or reuse, if one already exists elsewhere in `@markuplint/ml-spec` / `@markuplint/html-spec`) a table of HTML LS's optional-end-tag elements and their omission conditions (e.g. `<li>`: immediately followed by another `<li>` or the end of its parent; `<p>`: immediately followed by an element that isn't allowed as a child of `<p>` per the specific list in the spec, or the end of an ancestor that doesn't allow `<p>` — the full table is at <https://html.spec.whatwg.org/multipage/syntax.html#optional-tags>).
2. When `closeTag == null` for such an element, check whether the omission condition is actually met at this position (via next-sibling / parent-end inspection) before reporting.
3. Once the false-positive risk is eliminated, this rule's category and default severity can be reconsidered (out of scope for a first pass — the bench can add `severity: 'error'` as a scoped override without touching the user-facing default, following the pattern already used for `table-row-column-alignment` / `wai-aria-implicit-props` in `tests/external/bench/config.ts`).

## Files

- `packages/@markuplint/rules/src/end-tag/index.ts`
- `packages/@markuplint/rules/src/end-tag/index.spec.ts`
- `packages/@markuplint/rules/src/end-tag/README.md` / `README.ja.md`
- `tests/external/bench/config.ts` (bench enablement, after the fix)

## Bench impact

1 fixture (`html-syntax-picture-no-end-tag-novalid`) currently `nu-only`. Resolving this Issue and enabling the rule for the bench is expected to flip it to `match-error`. The omission-table fix itself has much broader impact on `end-tag`'s real-world accuracy beyond the bench.

**Comment by YusukeHirao** [comment]
Folding into #3943 — same fixture (`html-syntax-picture-no-end-tag-novalid`), and the analysis (end-tag rule's tag-omission blind spot) is more useful as context there than as a standalone Issue. See the comment on #3943 for the full writeup.
